engine/api/services.go

```diff
@@ -18,6 +19,40 @@ import (
 	"github.com/ovh/cds/sdk/log"
 )
 
+// killDeadServices must be run as a goroutine. It Deletes all dead workers
+func killDeadServices(ctx context.Context, dbFunc func() *gorp.DbMap) {
+	tick := time.NewTicker(30 * time.Second)
+	db := dbFunc()
+	for {
+		select {
+		case <-tick.C:
+			svcs, errdead := services.FindDeadServices(db, 3*60*time.Second)
+			if errdead != nil {
+				log.Error("killDeadServices> Unable to find dead services: %v", errdead)
+				continue
+			}
+			for i := range svcs {
+				log.Info("killDeadServices> Delete dead service: %v", &svcs[i].Name)
+				if err := services.Delete(db, &svcs[i]); err != nil {
+					log.Error("killDeadServices> Unable to find dead services: %v", errdead)
+					continue
+				}
+				if svcs[i].Type == "hatchery" {
+					log.Info("killDeadServices> Delete dead hatchery: %v", &svcs[i].Name)
+					if err := hatchery.DeleteHatcheryByName(db, svcs[i].Name); err != nil {
```

Comment:
we shouldn't have to delete manually the hatchery when we remove the service. There should be  a foreign key and a delete cascade